code/modules/integrated_electronics/_defines.dm

```diff
@@ -0,0 +1,128 @@
+#define COLOR_BLUE    			"#0000FF"
+#define COLOR_NAVY    			"#000080"
+#define COLOR_PINK    			"#FF00FF"
+#define COLOR_PURPLE  			"#800080"
+#define COLOR_ORANGE  			"#FF9900"
+#define COLOR_LUMINOL 			"#66FFFF"
+#define COLOR_BEIGE 			"#CEB689"
+#define COLOR_BLUE_GRAY 		"#6A97B0"
+#define COLOR_BROWN 			"#B19664"
+#define COLOR_DARK_BROWN 		"#917448"
+#define COLOR_DARK_ORANGE 		"#B95A00"
+#define COLOR_GREEN_GRAY 		"#8DAF6A"
+#define COLOR_RED_GRAY 			"#AA5F61"
+#define COLOR_PALE_BLUE_GRAY	"#8BBBD5"
+#define COLOR_PALE_GREEN_GRAY 	"#AED18B"
+#define COLOR_PALE_RED_GRAY		"#CC9090"
+#define COLOR_PALE_PURPLE_GRAY	"#BDA2BA"
+#define COLOR_PURPLE_GRAY 		"#A2819E"
+#define COLOR_RED_LIGHT         "#FF3333"
+#define COLOR_DEEP_SKY_BLUE     "#00e1ff"
+GLOBAL_LIST_EMPTY(all_integrated_circuits)
+
+/proc/iscrowbar(O)
+	if(istype(O, /obj/item/crowbar))
+		return 1
+	return 0
+
+
+
+/proc/initialize_integrated_circuits_list()
+	for(var/thing in typesof(/obj/item/integrated_circuit))
+		GLOB.all_integrated_circuits += new thing()
+
+/proc/between(var/low, var/middle, var/high)
+	return max(min(middle, high), low)
+
+/obj/item/integrated_circuit
+	name = "integrated circuit"
+	desc = "It's a tiny chip!  This one doesn't seem to do much, however."
+	icon = 'icons/obj/electronic_assemblies.dmi'
+	icon_state = "template"
+	w_class = WEIGHT_CLASS_TINY
+	var/obj/item/device/electronic_assembly/assembly = null // Reference to the assembly holding this circuit, if any.
+	var/extended_desc = null
+	var/list/inputs = list()
+	var/list/inputs_default = list()			// Assoc list which will fill a pin with data upon creation.  e.g. "2" = 0 will set input pin 2 to equal 0 instead of null.
+	var/list/outputs = list()
+	var/list/outputs_default = list()		// Ditto, for output.
+	var/list/activators = list()
+	var/next_use = 0 //Uses world.time
+	var/complexity = 1 				//This acts as a limitation on building machines, more resource-intensive components cost more 'space'.
+	var/size = null					//This acts as a limitation on building machines, bigger components cost more 'space'. -1 for size 0
+	var/cooldown_per_use = 9 // Circuits are limited in how many times they can be work()'d by this variable.
+	var/power_draw_per_use = 0 		// How much power is drawn when work()'d.
+	var/power_draw_idle = 0			// How much power is drawn when doing nothing.
+	var/spawn_flags = null			// Used for world initializing, see the #defines above.
+	var/category_text = "NO CATEGORY THIS IS A BUG"	// To show up on circuit printer, and perhaps other places.
+	var/removable = TRUE 			// Determines if a circuit is removable from the assembly.
+	var/displayed_name = ""
+	var/allow_multitool = 1			// Allows additional multitool functionality
+									// Used as a global var, (Do not set manually in children).
+proc/get_random_colour(var/simple, var/lower=0, var/upper=255)
+	var/colour
+	if(simple)
+		colour = pick(list("FF0000","FF7F00","FFFF00","00FF00","0000FF","4B0082","8F00FF"))
```

Comment:
A list in GLOB of all the color defines above would probably be best to pick from

Reply:
global list for 7 colors? Do you think so?

Reply:
Only if you end up keeping the color defines above

Reply:
It would be a global list of all the color defines, not just these colors.

If you only want these colors have a static list instead

Reply:
i think we already have one in fact

Reply:
deleted.